### DISCUSSION TITLE
[QUESTION] How to use import.meta.glob to dynamically import translations

### DISCUSSION THREAD

**Discussion by Red-Asuka** [nuxt-modules/i18n#2218] [Q&A] [answered]
### Environment

Operating System: MacOS
Node Version: v20.3.1
Nuxt Version: 3.6.1
Nitro Version: 2.5.2
Package Manager: pnpm@8.6.5
Builder: vite

### Reproduction

https://stackblitz.com/edit/github-vjsmyp?file=lang%2Fen-US.ts

### Describe the bug

In Nuxt2, I use `require.context` to dynamically import translations, and it works well. Now I'm trying to upgrade the project to Nuxt3 and use `import.meta.glob`. When modifying the code under dev mode, it shows that it works fine after hot update. However, when running dev or build again, it displays an ERROR `'unknown' type in '/Users/xxx/lang/en-US.ts'`.

Is there any way to dynamically import translations?

<img width="651" alt="image" src="https://github.com/nuxt-modules/i18n/assets/38158783/a557c7e2-a81f-4d3f-aca5-1ebc675d95db">

**Comment by kazupon** [comment]
This is a question, so we will move to github disucuttions.

**Comment by kazupon** [comment]
You can use lazy loading 
See the docs here:
https://v8.i18n.nuxtjs.org/guide/lazy-load-translations

**Comment by Red-Asuka** [comment]
Due to the large amount of text on my website, I have divided the text based on pages. Using `files` requires manually maintaining a list of files, which needs to be modified every time a new page is added. Is there a way to specify importing all files in a specific folder, thus reducing the need for manual updates?

**Reply by Guito** [reply]
I did the following in the setup of the page / layout:
```
const file = 'example_en.js' // This can be created using your locale
const translations = import.meta.glob(`~/assets/locales/*.js`, {import: 'default'}); 
const currentLocaleTranslation = await translations[`/assets/locales/${file}`]()
mergeLocaleMessage(locale.value, currentLocaleTranslation)
```

**Reply by Red-Asuka** [reply, marked as answer]
@Guito Thank you for your reply, I finally set it up this way and achieved the same effect.

`/lang/en-US.ts`

```ts
export default defineI18nLocale(() => {
  const importAllTranslations = import.meta.glob(
    '../lang/en-US/*.ts',
    {
      eager: true,
      import: 'default',
    },
  )
  const combinedTranslations = Object.values(importAllTranslations).reduce(
    (acc: any, cur: any) => Object.assign(acc, cur),
    {},
  )
  return combinedTranslations
})
```

`i18n config`

```ts
lazy: true,
langDir: 'lang/',
defaultLocale: 'en',
locales: [
    {
      code: 'en',
      name: 'English',
      iso: 'en',
      file: 'en-US.ts',
    },
]
```
